File: plot/src/main/java/net/fwitz/math/main/escape/halley/HalleysMethodZCubedMinus2ZMinus5.java
package net.fwitz.math.main.escape.halley;

public class HalleysMethodZCubedMinus2ZMinus5 {
    public static void main(String[] args) {
        HalleysMethodPlot.render(
                "z^3 -2z - 5  (Halley's Method)",
                z -> z.pow3().minus(z.times(2)).minus(5),
                z -> z.pow2().times(3).minus(2),
                z -> z.times(6));
    }
}

File: main/src/main/kotlin/net/fwitz/math/main/binary/escape/halley/HalleysMethodZCubedMinus2ZMinus5.kt
package net.fwitz.math.main.binary.escape.halley

object HalleysMethodZCubedMinus2ZMinus5 {
    @JvmStatic
    fun main(args: Array<String>) = HalleysMethodPlot.render(
        name = "z^3 -2z - 5  (Halley's Method)",
        f = { z -> z.pow3 - z * 2 - 5 },
        df = { z -> z.pow3 * 3 - 2 },
        d2f = { z -> z * 6 }
    )
}